evaluation/evaluation.py

```diff
@@ -0,0 +1,190 @@
+import json
+import time
+from typing import Any
+from dataclasses import dataclass
+
+import vertexai
+import vertexai.preview.generative_models as generative_models
+
+from google.auth import default
+from vertexai.generative_models import GenerativeModel
+
+
+credentials, _ = default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
+
+GCP_PROJECT_ID = "exemplary-cycle-195718"
+GCP_LLM_LOCATION = "us-east1"  # NB: Gemini is not available in europe-west2 (yet?)
+CURRENT_MODEL = "gemini-1.5-pro-preview-0514"
+
+
+class ProcessingException(Exception):
+    def __init__(self, *args: object) -> None:
+        super().__init__(*args)
+
+
+class ParsingException(Exception):
+    def __init__(self, *args: object) -> None:
+        super().__init__(*args)
+
+
+@dataclass(frozen=True)
+class ProviderOptions:
+    id: str | None
+    config: dict[str, Any] | None
+
+    @staticmethod
+    def from_dict(dictionary: dict[str, Any]) -> "ProviderOptions":
+        return ProviderOptions(
+            id=dictionary.get("id", None),
+            config=dictionary.get("config", {}),
+        )
+
+    def to_dict(self) -> dict[str, Any]:
+        return {"id": self.id, "config": self.config}
+
+
+@dataclass(frozen=True)
+class CallApiContextParams:
+    vars: dict[str, str]
+
+    @staticmethod
+    def from_dict(dictionary: dict[str, Any]) -> "CallApiContextParams":
+        return CallApiContextParams(vars=dictionary["vars"])
+
+    def to_dict(self) -> dict[str, Any]:
+        return {"vars": self.vars}
+
+
+@dataclass(frozen=True)
+class TokenUsage:
+    total: int
+    prompt: int
+    completion: int
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "total": self.total,
+            "prompt": self.prompt,
+            "completion": self.completion,
+        }
+
+
+@dataclass(frozen=True)
+class ProviderResponse:
+    output: str | dict[str, Any] | None = None
+    error: str | None = None
+    tokenUsage: TokenUsage | None = None
+    cost: float | None = None
+    cached: bool | None = None
+    logProbs: list[float] | None = None
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "output": self.output,
+            "error": self.error,
+            "tokenUsage": (
+                self.tokenUsage.to_dict() if self.tokenUsage is not None else None
+            ),
+            "cost": self.cost,
+            "cached": self.cached,
+            "logProbs": self.logProbs,
+        }
+
+
+def load_model() -> GenerativeModel:
+    vertexai.init(project=GCP_PROJECT_ID, location=GCP_LLM_LOCATION)
+    return GenerativeModel(model_name=CURRENT_MODEL)
+
+
+def parse_model_json_output(model_output: str) -> list[dict[str, str]]:
```

Comment:
is this function supposed to be unused?

Reply:
Good spot. I wasn't using it because currently the only test is "is it proper json?". I wasn't sure if we should do the parsing in the evaluation script or not. I suppose it depends if we care if the output is actually json only, or if we don't mind so long as there's json in there.